### DISCUSSION TITLE
Matching result data in multiple requests with dsl doesn't work

### DISCUSSION THREAD

**Discussion by faf-xff** [projectdiscovery/nuclei#2170] [Q&A] [answered]
Request condition allows checking for condition between multiple requests for writing complex checks and exploits involving multiple HTTP request to complete the exploit chain.

example:

templatePath:  cves/2016/CVE-2016-3088.yaml

<img width="620" alt="image" src="https://user-images.githubusercontent.com/41617530/173799246-0634590a-c158-4717-aa5a-f7996485c58b.png">

After the actual test, the match did not succeed. In the final match, there is no data requested for the first time in the data.

<img width="997" alt="image" src="https://user-images.githubusercontent.com/41617530/173800225-e6bf89f3-c9c7-4e21-8c64-4c3c2a3d0372.png">

**Comment by ehsandeep** [comment, marked as answer]
@guanyufen123 thanks for flagging this, it is now fixed here - https://github.com/projectdiscovery/nuclei-templates/pull/4594, it was not working as [separate](https://github.com/projectdiscovery/nuclei-templates/pull/4594/files#diff-ccc734e4af0917de912660f4ce54a59a526aee213734b01cb5ef7560a80e83dbL28) raw block was used for both requests which caused an issue with req-condition dsl.
